### DISCUSSION TITLE
CI Feature: Implement pull request preview environments

### DISCUSSION THREAD

**Discussion by waveywaves** [rabbitmq/rabbitmq-server#6706] [Ideas]
I would like to support RabbitMQ by implementing [Uffizzi](https://github.com/UffizziCloud/uffizzi) preview environments.
Disclaimer: I work on [Uffizzi](https://github.com/UffizziCloud/uffizzi).

Uffizzi is a Open Source full stack previews engine and our platform is available completely free for RabbitMQ (and all open source projects). This will provide maintainers with preview environments of every PR in the cloud, which enables faster iterations and reduces time to merge. You can see the open source repos which are currently using Uffizzi over [here](https://uffizzi.notion.site/)

Uffizzi is purpose-built for the task of previewing PRs and it integrates with your workflow to deploy preview environments in the background without any manual steps for maintainers or contributors. We can go ahead and create an Initial PoC for you right away if you think there is value in this proposal.

- [ ] Initial PoC

**Comment by michaelklishin** [comment]
@waveywaves thank you for the offer. We'd consider using your service but in the foreseeable future, all of our build system time budget will be spent on Bazel, BuildBuddy and GitHub Actions. At some point we will take a look at Uffizzi and see where it would fit.

**Reply by waveywaves** [reply]
We have a GitHub actions workflow for uffizzi which can fit into yours. I can create a PoC for you if that works.
